client/lib/media/utils/create-transient-media-id.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import impureLodash from 'lib/impure-lodash';
+
+const { uniqueId } = impureLodash;
+
+/**
+ * Returns an ID for transient media items. To be consistent in creating
+ * transient media IDs they are all prefixed with 'media-'
+ *
+ * @param {string} moreSpecifiedPrefix can be used to further specify the prefix
+ */
+function createTransientMediaId( moreSpecifiedPrefix = '' ) {
```

Comment:
I think `moreSpecific` sounds more natural than `moreSpecified`, but hey, I'm not a natural English speaker, so take this with a grain of salt 🧂 😉

Reply:
`moreSpecific` sounds better to me, although I think it's more or less the same.. not a native speaker as well 😉 I'll change it!